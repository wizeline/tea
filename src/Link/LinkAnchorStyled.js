// @flow

import styled from 'styled-components';

const LinkAnchorStyled = styled.a`
  text-decoration: none !important;
`;

export default LinkAnchorStyled;
