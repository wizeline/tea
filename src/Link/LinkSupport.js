// @flow

import { css } from 'styled-components';
import type { LinkProps } from './LinkTypes';
import { supportBooleanNameGroup } from '../utils';
import { getTextStyleSupport } from '../Text/TextSupport';

const priorityPropName = 'priority';
const supportedPriorities = ['b50', 'b100', 'b150', 'h300'];
const defaultPriority = 'b100';

const getLinkPropSupport = (props: LinkProps) => {
  const { children, ...otherProps } = props;
  const priority = supportBooleanNameGroup(
    priorityPropName,
    supportedPriorities,
    defaultPriority,
  )(otherProps);
  const linkAnchorStyledProps = {
    ...priority,
  };
  return {
    children,
    linkAnchorStyledProps,
  };
};

const getLinkStyleSupport = (props: LinkProps) => {
  const color = props.theme ? props.theme.textLink : null;
  const colorOverride = color ? `color: ${color};` : '';
  return css`
    ${getTextStyleSupport};
    ${colorOverride};
  `;
};

export { getLinkPropSupport, getLinkStyleSupport };
