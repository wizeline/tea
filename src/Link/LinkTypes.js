// @flow

import * as React from 'react';

export type LinkProps = {
  children?: React.Node,
  b50?: boolean,
  b100?: boolean,
  b150?: boolean,
  h300?: boolean,
  href?: string,
  theme?: Object,
};
