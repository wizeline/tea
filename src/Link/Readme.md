 
 ```jsx
  <div>
    Default: <Link href="https://wizelineroadmap.com/" >Wizeline Roadmap</Link>
    <br />B50: <Link b50 href="https://wizelineroadmap.com/" >Wizeline Roadmap</Link>
    <br />B100: <Link b100 href="https://wizelineroadmap.com/" >Wizeline Roadmap</Link>
    <br />B150: <Link b150 href="https://wizelineroadmap.com/" >Wizeline Roadmap</Link>
    <br />H300: <Link h300 href="https://wizelineroadmap.com/" >Wizeline Roadmap</Link>
  </div>
```