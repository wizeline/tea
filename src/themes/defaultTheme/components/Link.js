// @flow

import colors from '../../styles/colors';
import otherConstants from '../constants';

export default {
  [otherConstants.TEXT_LINK]: colors.blueBase,
};
