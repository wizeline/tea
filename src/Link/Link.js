// @flow

import React from 'react';
import type { LinkProps } from './LinkTypes';
import LinkAnchorStyled from './LinkStyled';
import { getLinkPropSupport } from './LinkSupport';

const Link = (props: LinkProps) => {
  const { children, linkAnchorStyledProps } = getLinkPropSupport(props);
  return (
    <LinkAnchorStyled
      {...linkAnchorStyledProps}
      href={props.href}
      target={props.target}
      onClick={props.onClick || (() => {})}
    >
      {children}
    </LinkAnchorStyled>
  );
};

export default Link;
