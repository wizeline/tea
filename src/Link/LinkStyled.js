// @flow

/* eslint flowtype-errors/enforce-min-coverage: 0 */

import styled from 'styled-components';
import { getLinkStyleSupport } from './LinkSupport';

const LinkAnchorStyled = styled.a`
  text-decoration: none;
  ${getLinkStyleSupport};
`;

export default LinkAnchorStyled;
