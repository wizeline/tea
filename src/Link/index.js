// @flow

/* eslint flowtype-errors/enforce-min-coverage: 0 */
import Link from './Link';

export default Link;
