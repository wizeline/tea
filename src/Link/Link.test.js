import React from 'react';
import { shallow } from 'enzyme';
import Link from './Link';
import { shallowExpectToMatchSnapShot } from '../utils/testHelpers';

describe('Link', () => {
  it('functions correctly with children', () => {
    const childrenString = 'hello';
    const props = {
      href: 'http://wizeline.com/',
      target: '_blank',
      b150: true,
      children: childrenString,
    };
    shallowExpectToMatchSnapShot(<Link {...props} />);
  });

  it('functions correctly without children', () => {
    const props = {
      href: 'http://wizeline.com/',
      b150: true,
    };
    shallowExpectToMatchSnapShot(<Link {...props} />);
  });

  it('calls onClick function', () => {
    const onClickMocked = jest.fn();
    const props = {
      onClick: onClickMocked,
    };
    const link = shallow(<Link {...props} />);
    link.simulate('click');
    expect(onClickMocked.mock.calls.length).toEqual(1);
  });

  it('has default onClick function', () => {
    const props = {};
    const link = shallow(<Link {...props} />);
    link.simulate('click');
    expect(link.props().onClick).toBeTruthy();
  });
});
