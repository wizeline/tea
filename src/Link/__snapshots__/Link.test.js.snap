// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link functions correctly with children 1`] = `
<styled.a
  href="http://wizeline.com/"
  priority="b150"
  theme={
    Object {
      "textLink": "#50a5e7",
    }
  }
>
  hello
</styled.a>
`;

exports[`Link functions correctly without children 1`] = `
<styled.a
  href="http://wizeline.com/"
  priority="b150"
  theme={
    Object {
      "textLink": "#50a5e7",
    }
  }
/>
`;
